src/modules/git_state.rs

```diff
@@ -0,0 +1,165 @@
+use ansi_term::Color;
+use git2::{Repository, RepositoryState};
+use std::path::Path;
+
+use super::{Context, Module};
+
+/// Creates a module with the state of the git repository at the current directory
+///
+/// During a git operation it will show: REBASING, BISECTING, MERGING, etc.
+/// If the progress information is available (e.g. rebasing 3/10), it will show that too.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let repo_root = context.repo_root.as_ref()?;
+    let mut repository = Repository::open(repo_root).ok()?;
+    let state_description = get_state_description(&mut repository);
+
+    if let StateDescription::Clean = state_description {
+        return None;
+    }
+
+    let mut module = context.new_module("git_state")?;
+    module.get_prefix().set_value("(");
+    module.get_suffix().set_value(") ");
```

Comment:
Are the prefix and suffix set this way specifically because they're meant to be used right next to the other `git` modules? I think we usually go for something like " on " or " in " for defaults.

Reply:
Honestly, this choice was mostly to ensure that the prompt doesn't get too long. In the default prompt order, this comes directly after the branch name, so it usually makes sense it context.

That said, I'm happy to make changes if people think there are better options..?

Reply:
I think this is a fine compromise to make for now. I think at some point in the future, we're going to have to start adding shell adapter code that tells starship what size the terminal is, and let starship decide what gets sacrificed if the terminal is too small to fit everything.

(I also seem to recall properly handling SIGWINCH being a PITA, so I'm really looking forward to that...)